src/tables/ctran_data.py

```diff
@@ -106,6 +106,43 @@ def create_table(self, ctran_sample_path="assets/"):
         self._print("Done.")
         return True
 
+    #######################################################
+
+    # Query all data between date_from and date_to, dates
+    # NOTE: if there is no ctran_data table, this will not work, obviously.
+    # TODO confer with Sawyer over testing
+    def query_date_range(self, date_from, date_to):
+        if self._engine is None:
+            self._print("ERROR: self._engine is None, cannot continue.")
+            return None
+
+        df = None
+        sql = "".join(["SELECT * FROM ",
+                       self._schema,
+                       ".",
+                       self._table_name,
+                       " WHERE service_date BETWEEN '",
+                       date_from.strftime("%Y-%m-%d"),
+                       "' AND '",
+                       date_to.strftime("%Y-%m-%d"),
+                       "';"])
+        self._print(sql)
+        try:
+            df = pandas.read_sql(sql, self._engine, index_col=self._index_col)
+
+        except SQLAlchemyError as error:
+            print("SQLAclchemy:", error)
+            return None
+        except ValueError as error:
```

Comment:
I found out Pandas can also throw a KeyError on invalid index_col, so if you could change this from `ValueError` to `(KeyError, ValueError)` that would be preferred.